packages/contracts/contracts/L1/teleportr/TeleportrDeposit.sol

```diff
@@ -0,0 +1,140 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.10;
+
+contract TeleportrDeposit {
+    address private owner;
+    uint256 private maxDepositAmount;
+    uint256 private minDepositAmount;
+    uint256 private maxBalance;
+    bool private canReceiveDeposit;
+    uint256 public totalDeposits;
+
+    constructor(
+        uint256 _maxDepositAmount,
+        uint256 _minDepositAmount,
+        uint256 _maxBalance,
+        bool _canReceiveDeposit
+    ) {
+        owner = msg.sender;
+        maxDepositAmount = _maxDepositAmount;
+        minDepositAmount = _minDepositAmount;
+        maxBalance = _maxBalance;
+        canReceiveDeposit = _canReceiveDeposit;
+        totalDeposits = 0;
+        emit OwnerSet(address(0), msg.sender);
+        emit MaxDepositAmountSet(0, _maxDepositAmount);
+        emit MinDepositAmountSet(0, _minDepositAmount);
+        emit MaxBalanceSet(0, _maxBalance);
+        emit CanReceiveDepositSet(_canReceiveDeposit);
```

Comment:
Do we really need all these events? Is anyone actually watching for them?

Reply:
It is a lot of events but it does follow best practices re: allowing an offchain actor to observe all events and recompute the state of the contract

Reply:
I removed the events, but if we want to add them back as per @tynes comment I can do that

Reply:
Honestly I think @tynes makes a good point.
At least we can remove `CanReceiveDepositSet` now.

Reply:
This contract has a backdoor in it where all the money can be drained so we should make it transparent in its operations. We should definitely have an event for `withdrawBalance` - it could help with ensuring security of the key we are using